components/dropdown/dropdown-content-mixin.js

```diff
@@ -0,0 +1,556 @@
+				spaceAround.right = 0;
+			}
+
+			const centerDelta = contentRect.width - targetRect.width;
+			const contentXAdjustment = centerDelta / 2;
+			if (centerDelta > 0) {
+				// content wider than target, so slide left/right as needed
+				if (!this.__isRTL) {
+					if (spaceAround.left > contentXAdjustment && spaceAround.right > contentXAdjustment) {
+						// center with target
+						position.style.left = `${contentXAdjustment * -1}px`;
+					} else if (spaceAround.left < contentXAdjustment) {
+						// slide content right (not enough space to center)
+						position.style.left = `${(spaceAround.left) * -1}px`;
+					} else if (spaceAround.right < contentXAdjustment) {
+						// slide content left (not enough space to center)
+						position.style.left = `${((contentRect.width - targetRect.width) * -1) + spaceAround.right}px`;
+					}
+				} else {
+					if (spaceAround.left > contentXAdjustment && spaceAround.right > contentXAdjustment) {
+						// center with target
+						position.style.right = `${contentXAdjustment * -1}px`;
+					} else if (spaceAround.left < contentXAdjustment) {
+						// slide content right (not enough space to center)
+						position.style.right = `${((contentRect.width - targetRect.width) * -1) + spaceAround.left}px`;
+					} else if (spaceAround.right < contentXAdjustment) {
+						// slide content left (not enough space to center)
+						position.style.right = `${(spaceAround.right) * -1}px`;
+					}
+				}
+			} else {
+				// content narrower than target, so slide in
+				if (!this.__isRTL) {
+					position.style.left = `${contentXAdjustment * -1}px`;
+				} else {
+					position.style.right = `${contentXAdjustment * -1}px`;
+				}
+			}
+
+			if (!this.noAutoFit && maxHeight && maxHeight > 0) {
+				content.style.maxHeight = `${maxHeight}px`;
+				this.__toggleOverflowY(contentRect.height > maxHeight);
+			}
+
+			this.dispatchEvent(new CustomEvent('d2l-dropdown-position', { bubbles: true, composed: true }));
+		};
+
+		/* don't let dropdown content horizontally overflow viewport */
+		widthContainer.style.width = '';
+		content.style.width = '';
+
+		let width = window.innerWidth - 40;
+		if (width > content.scrollWidth) {
+			width = content.scrollWidth;
+		}
+
+		/* add 2 to width since scrollWidth does not include border */
+		widthContainer.style.width = `${width + 20}px`;
+		/* set width of content too so IE will render scroll inside border */
+		content.style.width = `${width + 18}px`;
+
+		adjustPosition();
+	}
+
+	__toggleOverflowY(isOverflowing) {
+		if (!this.__content || !this.__content.style || !this.__content.style.maxHeight) {
+			return;
+		}
+
+		const maxHeight = parseInt(this.__content.style.maxHeight, 10);
+		if (!maxHeight) {
+			return;
+		}
+
+		if (isOverflowing || this.__content.scrollHeight > maxHeight) {
+			this.__content.style.overflowY = 'auto';
+		} else {
+			/* needed for IE */
+			this.__content.style.overflowY = 'hidden';
+		}
+	}
+
+	__toggleScrollStyles() {

```

Comment:
Same thing for this method - I think we need to refactor slightly so that we update properties, and then these classes get added/removed via `render`.  Lit will batch these updates and they will all happen in its render phase.